server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletStateStore.java

```diff
@@ -52,6 +61,22 @@ public interface TabletStateStore extends Iterable<TabletLocationState> {
   @Override
   ClosableIterator<TabletLocationState> iterator();
 
+  /**
+   * Create a stream of TabletLocationState that automatically closes the underlying iterator.
+   */
+  default Stream<TabletLocationState> stream() {
+    ClosableIterator<TabletLocationState> iterator = this.iterator();
+    return StreamSupport
+        .stream(Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false)
+        .onClose(() -> {
+          try {
+            iterator.close();
+          } catch (IOException e) {
+            log.warn("Error closing iterator", e);
```

Comment:
Probably best to let it out as an unchecked exception

```suggestion
            throw new UncheckedIOException(e);
```